src/openeo_gfmap/inference/inference_models.py

```diff
@@ -0,0 +1,51 @@
+"""Inference functionalities. Such as a base class to assist the implementation
+of inference models on an UDF.
+"""
+
+from abc import ABC, abstractmethod
+
+import xarray as xr
+
+from openeo_gfmap.features.feature_extractor import EPSG_HARMONIZED_NAME
+
+REQUIRED_IMPORTS = """
+import inspect
+from abc import ABC, abstractmethod
+
+import openeo
+from openeo.udf import XarrayDataCube
+from openeo.udf.run_code import execute_local_udf
+from openeo.udf.udf_data import UdfData
+
+from openeo_gfmap.features.feature_extractor import EPSG_HARMONIZED_NAME
+
+import xarray as xr
+import numpy as np
+
+from typing import Union
+"""
+
+
+class ModelInference(ABC):
+    """Base class for all model inference UDFs. It provides some common
+    methods and attributes to be used by other model inference classes.
+    """
+
+    @abstractmethod
+    def _import_dependencies(self):
```

Comment:
maybe I'm misunderstanding, but I don't think it makes sense to import dependencies in a method like this.

note that local imports resolve to local variables that are unreachable from the outside

Reply:
This was resolved for feature extractors UDFs but wasn't fixed for inference UDFs, anyways they should be part of a different PR